src/promptflow/promptflow/_core/tools_manager.py

```diff
@@ -308,8 +335,8 @@ def load_tool_for_script_node(self, node: Node) -> Tuple[Callable, Tool]:
         m = load_python_module_from_file(self._working_dir / path)
         if m is None:
             raise CustomToolSourceLoadError(f"Cannot load module from {path}.")
-        f = collect_tool_function_in_module(m)
-        return f, _parse_tool_from_function(f)
+        f, initialize_inputs = collect_tool_function_with_init_inputs_in_module(m)
+        return m, _parse_tool_from_function(f, initialize_inputs)
```

Comment:
Why do we return module here?

Reply:
Because we need to load function from module. I have removed the module here and use a real module name when load it rather than a hard code name "__pf__main__", in order to load module based on module name latter.